File: algorithms/1491.average-salary-excluding-the-minimum-and-maximum-salary/solution.go
func average(salary []int) float64 {
	min := salary[0]
	max := 0
	l := len(salary)
	sum := 0

	for _, s := range salary {
		if s < min {
			min = s
		}
		if s > max {
			max = s
		}
		sum += s
	}

	return float64(sum-min-max) / float64(l-2)
}

File: algorithms/1491.average-salary-excluding-the-minimum-and-maximum-salary/solution.ts
function average(salary: number[]): number {
  let min = salary[0];
  let max = 0;
  let sum = 0;

  for (const s of salary) {
    if (s < min) {
      min = s;
    }
    if (s > max) {
      max = s;
    }
    sum += s;
  }

  return (sum - min - max) / (salary.length - 2);
}
